scripts/build_test.py

```diff
@@ -36,6 +115,211 @@ def test_insert_hash(self):
+        """
+        random_filepaths = [
+            os.path.join(build.THIRD_PARTY_GENERATED_DEV_DIR, 'random1.js')]
+        # pylint: disable=protected-access
+        with self.assertRaises(ValueError) as fileNotExist:
+            build._ensure_files_exist(random_filepaths)
+        # pylint: enable=protected-access
+        # Exception will be raised at first file determined to be non-existent.
+        self.assertTrue(
+            ('File %s does not exist.') % random_filepaths[0] in
+            fileNotExist.exception)
+
+    def test_ensure_fonts_exist(self):
+        """Test _ensure_fonts_exist raises exception with a non-existent
+        font filepath.
+        """
+        random_fontpaths = [
+            os.path.join(
+                build.THIRD_PARTY_GENERATED_DEV_DIR, 'fontunimpressive.txt')]
+
+        with self.assertRaises(ValueError) as fileNotExist:
+            # pylint: disable=protected-access
+            build._ensure_files_exist(random_fontpaths)
+            # pylint: enable=protected-access
+        # Exception will be raised at first file determined to be non-existent.
+        self.assertTrue(
+            ('File %s does not exist.') % random_fontpaths[0] in
+            fileNotExist.exception)
+
+    def test_get_file_count(self):
+        """Test get_file_count to return the correct number of files, minus
+        ignored files.
+        """
+        all_inclusive_file_count = 0
+        for _, _, files in os.walk(build.EXTENSIONS_DEV_DIR):
+            all_inclusive_file_count += len(files)
+        ignored_file_count = 0
+        for _, _, files in os.walk(build.EXTENSIONS_DEV_DIR):
+            for filename in files:
+                if any(filename.endswith(p)
+                       for p in build.FILE_EXTENSIONS_TO_IGNORE):
+                    ignored_file_count += 1
+        self.assertEqual(
+            all_inclusive_file_count - ignored_file_count,
+            build.get_file_count(build.EXTENSIONS_DEV_DIR))
+
+    def test_compare_file_count(self):
+        """Test _compare_file_count to raise exception when there is a
+        mismatched file count between 2 dirs.
+        """
+        with self.assertRaises(ValueError) as incorrectFileCount:
+            # pylint: disable=protected-access
+            build._compare_file_count(
+                build.EXTENSIONS_DEV_DIR, build.TEMPLATES_DEV_DIR_CORE)
+        # pylint: enable=protected-access
+        source_dir_file_count = build.get_file_count(build.EXTENSIONS_DEV_DIR)
+        target_dir_file_count = build.get_file_count(
+            build.TEMPLATES_DEV_DIR_CORE)
+        self.assertTrue(
+            ('%s files in source dir != %s files in target dir.') % (
+                source_dir_file_count, target_dir_file_count) in
+            incorrectFileCount.exception)
+
+    def test_match_directory_with_hashes(self):
+        """Test _match_directory_with_hashes to raise exception:
+            1)when there is an empty hash dict.
```

Comment:
Nit: "When" and maybe add space after the `1)`

Reply:
Yep. Weird stuff going on here with the docstring. Fixed!